contracts/contracts/CalldataEncryption.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+import {Subcall} from "./Subcall.sol";
+import {Sapphire} from "./Sapphire.sol";
+import {CBOR_encodeBytes, CBOR_encodeUint} from "./CBOR.sol";
+
+library CalldataEncryption {
+    function _deriveKey(
+        bytes32 in_peerPublicKey,
+        Sapphire.Curve25519SecretKey in_x25519_secret
+    ) internal view returns (bytes32) {
+        return
+            Sapphire.deriveSymmetricKey(
+                Sapphire.Curve25519PublicKey.wrap(in_peerPublicKey),
+                in_x25519_secret
+            );
+    }
+
+    function _encryptInner(
+        bytes memory in_data,
+        Sapphire.Curve25519SecretKey in_x25519_secret,
+        bytes15 nonce,
+        bytes32 peerPublicKey
+    ) internal view returns (bytes memory out_encrypted) {
+        bytes memory plaintextEnvelope = abi.encodePacked(
+            hex"a1", // map(1)
+            hex"64", //     text(4) "body"
```

Comment:
```suggestion
            hex"64", // text(4) "body"
```

Reply:
This is formatted according to the CBOR debug output I'm using throughout this PR.
It's vaguely hierarchial.
Unfortunately the Solidity prettifier is butchering any attempt other than comments to make it readable.